### ISSUE TITLE
Dialogs not visible

### ISSUE THREAD

**Issue by DirkHeinke** [qtile/qtile#809]
I have a four screen setup with on screen on top, and three screens on the bottom with the right screen rotated. Like this:
```
  -
- - |
```

Somehow I don't see dialogs. For example the "Save as... " in Chrome does not show up, but if I don't move my mouse I have focus on it. So I can `toggle_floating()` on that window and it shows up.

It seems like the dialog opens somewhere, where no screen is:

```
{'float_info': {'h': 1048, 'w': 958, 'x': -1920, 'y': -1080},
 'floating': True,
 'fullscreen': False,
 'group': '1',
 'height': 1048,
 'id': 6300169,
 'maximized': False,
 'minimized': False,
 'name': 'Save File',
 'width': 958,
 'x': 0,
 'y': 0}
```

Is my assumption correct and what else can I provide to help to fix this bug?

**Comment by tych0** [comment]
On Tue, Nov 17, 2015 at 07:44:47AM -0800, Dirk Heinke wrote:
> I have a four screen setup with on screen on top, and three screens on the bottom with the right screen rotated. Like this:
> ```
>   -
> - - |
> ```
> 
> Somehow I don't see dialogs. For example the "Save as... " in Chrome does not show up, but if I don't move my mouse I have focus on it. So I can `toggle_floating()` on that window and it shows up.
> 
> It seems like the dialog opens somewhere, where no screen is:
> 
> ```
> {'float_info': {'h': 1048, 'w': 958, 'x': -1920, 'y': -1080},
>  'floating': True,
>  'fullscreen': False,
>  'group': '1',
>  'height': 1048,
>  'id': 6300169,
>  'maximized': False,
>  'minimized': False,
>  'name': 'Save File',
>  'width': 958,
>  'x': 0,
>  'y': 0}
> ```
> 
> Is my assumption correct and what else can I provide to help to fix this bug?

I'll bet this is related to #538 (#726, etc.). I remember looking at
this a while ago and not having any brilliant insights. Probably the
floating code just needs to ask which screen actually has focus before
it creates the window.

**Comment by flacjacket** [comment]
It's odd that it's putting the window totally off the screen, at negative x and y, I haven't seen that, usually it is a problem where it places the dialog box on the wrong screen, but still visible. But yes, definitely related to our floating window placement issues.

**Comment by flacjacket** [comment]
I just fixed up a lot of floating stuff, this should be covered by it. Let us know if you still have any problems!

**Comment by DirkHeinke** [comment]
Hi,

I just updated to 0.10.5 and I still have the problem, that dialogs open somewhere but not on the correct screen (or invisible).

Example: Chrome opened on my monitor in the middle. I open the save dialog and it appears on my top screen (first info). If I have chrome on my left monitor, it opens somewhere where I don't see it.
```
{'float_info': {'h': 1048, 'w': 958, 'x': 0, 'y': -1080},
 'floating': True,
 'fullscreen': False,
 'group': '1',
 'height': 1048,
 'id': 16777580,
 'maximized': False,
 'minimized': False,
 'name': 'Save File',
 'width': 958,
 'x': 1920,
 'y': 0}
```

```
{'float_info': {'h': 1048, 'w': 958, 'x': 0, 'y': -1080},
 'floating': True,
 'fullscreen': False,
 'group': '1',
 'height': 1048,
 'id': 16777649,
 'maximized': False,
 'minimized': False,
 'name': 'Save File',
 'width': 958,
 'x': 0,
 'y': 0}
```

Any ideas, data you need?

**Comment by flacjacket** [comment]
We wound up not including the fixes in 0.10.5 so we could do some
additional testing, but it currently is only in the git version. We haven't
had any bug reports with it recently, so we could cut another point release
with the floating fixes in it.
